__tests__/sagas/networkSettings.test.ts

```diff
@@ -0,0 +1,260 @@
+import * as nsSaga from '../../src/sagas/networkSettings';
+import * as sagaHelpers from '../../src/sagas/helpers';
+import { all, effectTypes, fork } from 'redux-saga/effects';
+import createSagaMiddleware, { END, runSaga } from 'redux-saga';
+import { applyMiddleware, createStore } from 'redux';
+import { reducer } from '../../src/reducer';
+import { networkSettingsUpdate, networkSettingsUpdateSuccess, reloadWalletRequested, types } from '../../src/actions';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { networkSettingsKey } from '../../src/constants';
+import { STORE } from '../../src/store';
+import { config } from '@hathor/wallet-lib';
+
+beforeEach(() => {
+    jest.clearAllMocks();
+});
+
+describe('updateNetworkSettings', () => {
+    beforeAll(() => {
+        jest.spyOn(config, 'getExplorerServiceBaseUrl').mockReturnValue('');
+        jest.spyOn(config, 'getServerUrl');
+        jest.spyOn(config, 'getWalletServiceBaseUrl').mockReturnValue('');
+        jest.spyOn(config, 'getWalletServiceBaseWsUrl').mockReturnValue('');;
+    })
+
+    it('should fail with invalid input', async () => {
+        const actual: string[] = [];
+        // simulates saga cluster in sagas/index.js
+        const sagas = [nsSaga.saga];
+        function* defaultSaga() {
+            yield all(
+                sagas.map((saga) => fork(saga))
+            );
+        }
+
+        // effect middleware used to control effect behavior
+        const effectMiddleware = next => effect => {
+            if (effect.type === effectTypes.PUT && effect.payload.action.type === types.NETWORKSETTINGS_UPDATE_FAILURE) {
+                actual.push('failed')
+            }
+            return next(effect);
+        }
+
+        const middleware = createSagaMiddleware({ effectMiddlewares: [effectMiddleware] });
+        const store = createStore(reducer, {}, applyMiddleware(middleware));
+
+        const task = middleware.run(defaultSaga);
+
+        Promise.resolve()
```

Comment:
Why do you need this first `Promise.resolve()`?

Reply:
Look at this article ["Understanding setImmediate()"](https://nodejs.dev/en/learn/understanding-setimmediate/). To cut short I use it just to resume the promises before the end of the call stack, before the next event loop, which is possible the execution of another test. Without that, we risk a memory leak or jest zombie(open) handler. See caution in the jest ["Async/Await"](https://jestjs.io/docs/asynchronous#asyncawait).

Reply:
Great, thanks